project_material/models/stock_warehouse.py

```diff
@@ -0,0 +1,451 @@
+# © 2019 - today Numigi (tm) and all its contributors (https://bit.ly/numigiens)
+# License LGPL-3.0 or later (http://www.gnu.org/licenses/lgpl).
+
+from odoo import api, fields, models, SUPERUSER_ID, _
+
+
+ONE_STEP_KEY = "one_step"
+ONE_STEP_DESCRIPTION = _("Direct consumption from stocks (1 step)")
+
+
+class Warehouse(models.Model):
+    """Add consumption route to the warehouse."""
+
+    _inherit = 'stock.warehouse'
+
+    def _get_default_consumption_location_id(self):
+        return self.env.ref('stock.location_production', raise_if_not_found=False).id
+
+    consu_steps = fields.Selection([
+        (ONE_STEP_KEY, ONE_STEP_DESCRIPTION),
+    ], default=ONE_STEP_KEY)
+
+    def _has_one_step_consumption(self):
+        return self.consu_steps == ONE_STEP_KEY
+
+    consu_location_id = fields.Many2one(
+        'stock.location', 'Consumption Location',
+        domain=[('usage', '=', 'production')],
+        ondelete='restrict',
+        default=_get_default_consumption_location_id,
+    )
+
+    consu_type_id = fields.Many2one(
+        'stock.picking.type', 'Consumption Picking Type',
+        ondelete='restrict',
+    )
+
+    consu_return_type_id = fields.Many2one(
+        'stock.picking.type', 'Consumption Return Picking Type',
+        ondelete='restrict',
+    )
+
+    consu_route_id = fields.Many2one(
+        'stock.location.route', 'Consumption Route',
+        ondelete='restrict',
+    )
+
+    consu_mto_pull_id = fields.Many2one(
+        'procurement.rule', 'Consumption MTO Pull',
+        ondelete='restrict'
+    )
+
+    @api.model
+    def create(self, vals):
+        """When creating a new warehouse, create the consumption route.
+
+        Use sudo to prevent errors related to access rights.
+        """
+        warehouse = super().create(vals)
+        warehouse.sudo()._create_consumption_picking_types()
+        warehouse.sudo()._create_consumption_route()
+        warehouse.sudo()._create_consumption_mto_pull()
+        return warehouse
+
+    @api.multi
+    def write(self, vals):
+        """When changing the consumation steps, update the consumption route.
```

Comment:
Codacy found an issue: [Multi-line docstring summary should start at the second line (D213)](https://app.codacy.com/gh/Numigi/odoo-project-addons/pullRequest?prid=6119862)